File: frontend/src/app/models/cv-generation.models.ts
export interface StepStatus {
  step: number;
  name: string;
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled';
  started_at: string | null;
  completed_at: string | null;
  duration_ms: number | null;
  error: string | null;
}

export interface CvGenerationStatus {
  run_id: string;
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled';
  current_step: number;
  steps: StepStatus[];
  error_message: string | null;
  created_at: string;
  completed_at: string | null;
  cancelled_at: string | null;
}

export interface CvGenerationResult {
  run_id: string;
  extraction: any;
  search: any;
  optimization: any;
  render: any;
  delivery: any;
}

export interface CvGenerationSubmitResponse {
  run_id: string;
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
}

export interface CVProfile {
  id: string;
  title: string;
  summary?: string;
}

export interface EditCvSection {
  summary: string;
  experiences: EditExperience[];
  skills: string[];
  projects: EditProject[];
}

export interface EditExperience {
  company: string;
  title: string;
  description: string;
  bullets: string[];
}

export interface EditProject {
  name: string;
  description: string;
  url?: string;
}

export type InputTab = 'paste' | 'url' | 'file';
export type PageState = 'form' | 'progress' | 'results';

export const TONES = ['Confident', 'Warm', 'Technical', 'Concise'] as const;
export const LANGUAGES = [
  { code: 'en', label: 'English' },
  { code: 'fr', label: 'French' },
  { code: 'es', label: 'Spanish' },
  { code: 'de', label: 'German' },
  { code: 'nl', label: 'Dutch' },
  { code: 'it', label: 'Italian' },
  { code: 'pt', label: 'Portuguese' },
] as const;

export const TEMPLATES = [
  { id: 'default', label: 'Default' },
  { id: 'modern', label: 'Modern' },
  { id: 'classic', label: 'Classic' },
  { id: 'minimal', label: 'Minimal' },
] as const;

export const STEP_ICONS: Record<string, string> = {
  'Job Extraction': 'ti ti-file-search',
  'Profile Matching': 'ti ti-users',
  'CV Optimization': 'ti ti-sparkles',
  'Template Rendering': 'ti ti-template',
  'Email Delivery': 'ti ti-mail',
};

File: monolith/backend/Dto/CvGenerationModels.cs
using System.Text.Json.Serialization;

namespace CV_Generator.Dto;

public class StepStatusDto
{
    [JsonPropertyName("step")]
    public int Step { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("status")]
    public string Status { get; set; } = "pending";

    [JsonPropertyName("started_at")]
    public DateTime? StartedAt { get; set; }

    [JsonPropertyName("completed_at")]
    public DateTime? CompletedAt { get; set; }

    [JsonPropertyName("duration_ms")]
    public long? DurationMs { get; set; }

    [JsonPropertyName("error")]
    public string? Error { get; set; }
}

public class CvGenerationStatusResponse
{
    [JsonPropertyName("run_id")]
    public Guid RunId { get; set; }

    [JsonPropertyName("status")]
    public string Status { get; set; } = string.Empty;

    [JsonPropertyName("current_step")]
    public int CurrentStep { get; set; }

    [JsonPropertyName("steps")]
    public List<StepStatusDto> Steps { get; set; } = new();

    [JsonPropertyName("error_message")]
    public string? ErrorMessage { get; set; }

    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }

    [JsonPropertyName("completed_at")]
    public DateTime? CompletedAt { get; set; }

    [JsonPropertyName("cancelled_at")]
    public DateTime? CancelledAt { get; set; }
}

public class CvGenerationResultResponse
{
    [JsonPropertyName("run_id")]
    public Guid RunId { get; set; }

    [JsonPropertyName("extraction")]
    public object? Extraction { get; set; }

    [JsonPropertyName("search")]
    public object? Search { get; set; }

    [JsonPropertyName("optimization")]
    public object? Optimization { get; set; }

    [JsonPropertyName("render")]
    public object? Render { get; set; }

    [JsonPropertyName("delivery")]
    public object? Delivery { get; set; }
}

public class CvGenerationSubmitResponse
{
    [JsonPropertyName("run_id")]
    public Guid RunId { get; set; }
}
